### PR TITLE
Fix crash when clicking workspaces headers 

### LINKED ISSUE #3149: Crash when clicking workspaces headers System.InvalidCastException: Specified cast is not valid
## 💥 Crash Report
It's quite easy to reproduce.
1. Click the play button to start a new TE
2. Click the project icon on the toolbar, or type something to bring the project's list separated by workspaces
3. Click the workspace name.
4. 💥 

### 👥 Impact
This might have been happening for while. People might or might not be affected by this.
I'm not sure. I can't find it on AppCenter.

### 🛰 Link
???

### 📝  Exception & Stack Trace
`Exception type and message`
```cs
Unhandled Exception:
                         E  System.InvalidCastException: Specified cast is not valid.
                         E    at (wrapper castclass) System.Object.__castclass_with_cache(object,intptr,intptr)
                         E    at MvvmCross.Commands.MvxAsyncCommand`1[T].ExecuteAsyncImpl (System.Object parameter) [0x0
                            0016] in <2d1a0b0d64974565947e3547d3bf75e4>:0
                         E    at MvvmCross.Commands.MvxAsyncCommandBase+<ExecuteConcurrentAsync>d__17.MoveNext () [0x000
                            a7] in <2d1a0b0d64974565947e3547d3bf75e4>:0
                         E  --- End of stack trace from previous location where exception was thrown ---
                         E    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <c5010c907
                            d454d4494039e22802961e6>:0
                         E    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.
                            Task task) [0x0003e] in <c5010c907d454d4494039e22802961e6>:0
                         E    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (Sy
                            stem.Threading.Tasks.Task task) [0x00028] in <c5010c907d454d4494039e22802961e6>:0
                         E    at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task ta
                            sk) [0x00008] in <c5010c907d454d4494039e22802961e6>:0
                         E    at System.Runtime.CompilerServices.ConfiguredTaskAwaitable+ConfiguredTaskAwaiter.GetResult
                             () [0x00000] in <c5010c907d454d4494039e22802961e6>:0
                         E    at MvvmCross.Commands.MvxAsyncCommandBase+<ExecuteAsync>d__16.MoveNext () [0x00085] in <2d
                            1a0b0d64974565947e3547d3bf75e4>:0
                         E  --- End of stack trace from previous location where exception was thrown ---
                         E    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <c5010c907
                            d454d4494039e22802961e6>:0
                         E    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.
                            Task task) [0x0003e] in <c5010c907d454d4494039e22802961e6>:0
                         E    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (Sy
                            stem.Threading.Tasks.Task task) [0x00028] in <c5010c907d454d4494039e22802961e6>:0
                         E    at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task ta
                            sk) [0x00008] in <c5010c907d454d4494039e22802961e6>:0
                         E    at System.Runtime.CompilerServices.ConfiguredTaskAwaitable+ConfiguredTaskAwaiter.GetResult
                             () [0x00000] in <c5010c907d454d4494039e22802961e6>:0
                         E    at MvvmCross.Commands.MvxAsyncCommandBase+<Execute>d__14.MoveNext () [0x0009a] in <2d1a0b0
                            d64974565947e3547d3bf75e4>:0
                         E  --- End of stack trace from previous location where exception was thrown ---
                         E    at (wrapper dynamic-method) System.Object.303f7673-9b63-4f54-b572-3faa8a2b1d13(intptr,intp
                            tr)
                mono-rt  E  [ERROR] FATAL UNHANDLED EXCEPTION: System.InvalidCastException: Specified cast is not valid.
                         E    at (wrapper castclass) System.Object.__castclass_with_cache(object,intptr,intptr)
                         E    at MvvmCross.Commands.MvxAsyncCommand`1[T].ExecuteAsyncImpl (System.Object parameter) [0x0
                            0016] in <2d1a0b0d64974565947e3547d3bf75e4>:0
                         E    at MvvmCross.Commands.MvxAsyncCommandBase+<ExecuteConcurrentAsync>d__17.MoveNext () [0x000
                            a7] in <2d1a0b0d64974565947e3547d3bf75e4>:0
                         E  --- End of stack trace from previous location where exception was thrown ---
                         E    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <c5010c907
                            d454d4494039e22802961e6>:0
                         E    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.
                            Task task) [0x0003e] in <c5010c907d454d4494039e22802961e6>:0
                         E    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (Sy
                            stem.Threading.Tasks.Task task) [0x00028] in <c5010c907d454d4494039e22802961e6>:0
                         E    at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task ta
                            sk) [0x00008] in <c5010c907d454d4494039e22802961e6>:0
                         E    at System.Runtime.CompilerServices.ConfiguredTaskAwaitable+ConfiguredTaskAwaiter.GetResult
                             () [0x00000] in <c5010c907d454d4494039e22802961e6>:0
                         E    at MvvmCross.Commands.MvxAsyncCommandBase+<ExecuteAsync>d__16.MoveNext () [0x00085] in <2d
                            1a0b0d64974565947e3547d3bf75e4>:0
                         E  --- End of stack trace from previous location where exception was thrown ---
                         E    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <c5010c907
                            d454d4494039e22802961e6>:0
                         E    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.
                            Task task) [0x0003e] in <c5010c907d454d4494039e22802961e6>:0
                         E    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (Sy
                            stem.Threading.Tasks.Task task) [0x00028] in <c5010c907d454d4494039e22802961e6>:0
                         E    at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task ta
                            sk) [0x00008] in <c5010c907d454d4494039e22802961e6>:0
                         E    at System.Runtime.CompilerServices.ConfiguredTaskAwaitable+ConfiguredTaskAwaiter.GetResult
                             () [0x00000] in <c5010c907d454d4494039e22802961e6>:0
                         E    at MvvmCross.Commands.MvxAsyncCommandBase+<Execute>d__14.MoveNext () [0x0009a] in <2d1a0b0
                            d64974565947e3547d3bf75e4>:0
                         E  --- End of stack trace from previous location where exception was thrown ---
                         E    at (wrapper dynamic-method) System.Object.303f7673-9b63-4f54-b572-3faa8a2b1d13(intptr,intp
                            tr)
```

### PR DESCRIPTION
**Pull request by daividssilverio**
## What's this?
A crash fix to avoid `System.InvalidCastException: Specified cast is not valid` on clicks from the `CreateSuggestionGroupedTableViewSource` adapter.

### Relationships
Closes #3149

## Why do we want this?
Because the user might click the workspace headers, and it would be bad if the app crashed because of that.

## How is it done?
The views that have the crash in question are not rxfactored yet, so this PR adds another base MvxViewHolder that checks wether or not to run the click command when a view click happens.

### Why not another way?
Rxfactoring won't be that easy on these views. But yeah, that would be way better. I'm still in favour of having this easy fix for now.

### Side effects
Clicks on the ws headers won't execute any commands, but they weren't anyway.

## Review hints
Review by commit works well.
Try reproducing the crash on develop.
Try reproducing it again on this PR branch.

## :squid: Permissions
Anyone, go ahead. 🦑 it.

### CODE REVIEW

File: Toggl.Giskard/ViewHolders/SafeMvxRecyclerViewHolder.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+using System;
+using System.Windows.Input;
+using Android.Runtime;
+using Android.Views;
+using MvvmCross.Droid.Support.V7.RecyclerView;
+using MvvmCross.Platforms.Android.Binding.BindingContext;
+
+namespace Toggl.Giskard.ViewHolders
+{
+    public class GuardedMvxRecyclerViewHolder : MvxRecyclerViewHolder
```
Comment (kornelijepetak) [inline Toggl.Giskard/ViewHolders/SafeMvxRecyclerViewHolder.cs]: There is a name mismatch between the class and the file.

File: Toggl.Giskard/Adapters/CreateSuggestionSegmentedRecyclerAdapter.cs [hunk 2]
```diff
@@ -66,10 +66,11 @@ public override RecyclerView.ViewHolder OnCreateViewHolder(ViewGroup parent, int
         {
             var itemBindingContext = new MvxAndroidBindingContext(parent.Context, BindingContext.LayoutInflaterHolder);
             var inflatedView = InflateViewForHolder(parent, viewType, itemBindingContext);
-            var viewHolder = new MvxRecyclerViewHolder(inflatedView, itemBindingContext)
+            var viewHolder = new GuardedMvxRecyclerViewHolder(inflatedView, itemBindingContext)
             {
                 Click = viewType == SuggestCreationViewType ? CreateCommand : ItemClick,
                 LongClick = ItemLongClick,
+                ShouldRunCommand = item => !(item is TCollection)
```
Comment (kornelijepetak) [inline Toggl.Giskard/Adapters/CreateSuggestionSegmentedRecyclerAdapter.cs]: I would rather go with `CanExecute` to make it consistent with the similar command interfaces (such as `ICommand` for example).

**Comment by kornelijepetak** [comment]
@daividssilverio being away, I will assume this PR and do the requested changes. However, another review would be wise in this case. Summoning @dshr .

**Comment by dshr** [comment]
But yeah, rxfactoring this is going to be **hard**

**Pull request merged**